qiskit/dagcircuit/_dagnode.py

```
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+   Object to represent the information at a node in the DAGCircuit
+"""
+import copy
+
+
+class DAGNode:
+    """
+    Object to represent the information at a node in the DAGCircuit
+
+    It is used as the return value from *_nodes() functions and can
+    be supplied to functions that take a node.
+
+    """
+    def __init__(self, data_dict, nid=-1):
+        """ Create a node """
+        self._node_id = nid
+        self.data_dict = data_dict
+
+    @property
+    def type(self):
+        """ Returns a str which is the type of the node else None"""
+        return self.data_dict.get('type')
+
+    @property
+    def op(self):
+        """ Returns the Instruction object corresponding to the op for the node else None"""
+        return self.data_dict.get('op')
```

this method should raise an error if the node type is not 'op'.